cranelift/fuzzgen/src/function_generator.rs

```diff
@@ -1833,11 +1832,10 @@ where
                         let default = next_block;
 
                         let target_count = self.param(&self.config.jump_table_entries)?;
-                        let targets = arbitrary_vec(
-                            self.u,
-                            target_count,
-                            self.resources.forward_blocks_without_params(block),
-                        )?;
+                        let mut targets = Vec::with_capacity(target_count);
+                        for _ in 0..target_count {
+                            targets.push(self.generate_target_block(block)?);
+                        }
```

Comment:
I have a mild preference for the iterator chain approach here, mostly to avoid thinking about allocation capacity. A neat Rust trick is that `Result` implements `FromIterator`: if any element of the iterator is `Err` then that's returned, otherwise all the `Ok` values are collected into the underlying collection type. So I believe this should work:
```suggestion
                        let targets = (0..target_count).map(|_| self.generate_target_block(block)).collect()?;
```

Reply:
Very nice! I started with the iterator chain and went back to a for loop when I couldn't think of an elegant way to handle the `Result`. I'll give this a try :)